Microsoft.HealthVault.Fhir/Transformers/MedicationRequestToHealthVault.cs

```diff
@@ -0,0 +1,193 @@
+                    if (hasInstruction)
+                    {
+                        var instruction = medicationRequest.DosageInstruction.First(
+                         dosage => dosage.AdditionalInstruction.Any()).AdditionalInstruction.First();
+                        prescription.Instructions = instruction.ToCodableValue();//.GetCodableValue();
+                    }
+                }
+
+                if (medicationRequest.Substitution != null)
+                {
+                    prescription.Substitution = GetSubstitutionCode(medicationRequest, prescription);
+                }
+                hvMedication.Prescription = prescription;
+            }
+            else
+            {
+                throw new NotSupportedException($"{nameof(MedicationRequest)} needs to have an embedded Requester Agent");
+            }
+
+            return hvMedication;
+        }
+
+        private static int? GetDaysFromDuration(Hl7.Fhir.Model.Duration duration)
+        {
+            if (duration == null)
+            {
+                return null;
+            }
+            if (duration.Value.HasValue)
+            {
+                switch (duration.Code)
+                {
+                    case "day":
+                        return decimal.ToInt32(duration.Value.Value);
+                    case "week":
+                    case "month":
+                    case "year":
+                        double value = decimal.ToDouble(duration.Value.Value);
+                        var fromUnit = UnitsNet.Duration.ParseUnit(duration.Code);
+                        var unitsNetDuration = UnitsNet.Duration.From(value, fromUnit);
+                        return Convert.ToInt32(unitsNetDuration.Days);
+                }
+            }
+            throw new NotImplementedException();
+        }
+
+        private static CodableValue GetSubstitutionCode(MedicationRequest medicationRequest, Prescription prescription)
+        {
+            if (medicationRequest.Substitution.Allowed.HasValue)
+            {
+                switch (medicationRequest.Substitution.Allowed.Value)
+                {
+                    case true:
+                        return HealthVaultMedicationSubstitutionCodes.SubstitutionPermitted;
+                    case false:
+                        return HealthVaultMedicationSubstitutionCodes.DispenseAsWritten;
+                }
+            }
+            throw new NotImplementedException();
+        }
+
+        private static FhirMedication ExtractEmbeddedMedication(MedicationRequest medicationRequest)
+        {
+            switch (medicationRequest.Medication)
+            {
+                case ResourceReference medicationReference:
+                    if (!medicationReference.IsContainedReference)
+                    {
+                        throw new NotImplementedException();
+                    }
+                    var fhirMedication = medicationRequest.Contained.First(domainResource
```

Comment:
>First [](start = 69, length = 5)

FirstOrDefault to avoid an exception in case there is no match? You are returning a new FhirMedication in null case